contrib/build-ceph-base.sh

```diff
@@ -0,0 +1,92 @@
+#!/usr/bin/env bash
+set -euo pipefail
+
+# Allow running this script with the env var DRY_RUN="<something>" to do a dry run of the
+# script. Dry runs will output commands that they would have executed as info messages.
+
+SCRIPT_DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" >/dev/null && pwd )" || exit $?
+# shellcheck disable=SC1090  # sourcing from a variable here does indeed work
+source "${SCRIPT_DIR}/ceph-build-config.sh"
+
+# As a note, bash functions which return strings do so by echo'ing the result. When the function
+# is called like 'var="$(fxn)"', var will get the return string. Most of these function calls also
+# include 'var="$(fxn)" || exit $?'. Without the or-exit on the end, the script will not fail if
+# the inner subshell fails.
+
+
+#
+# Main
+#
+
+flavors_to_build="$(get_flavors_to_build "${ARCH}")" || exit $?
+
+echo ''
+info "BUILDING CEPH-${ARCH} IMAGES FOR FLAVORS:"
+info "  ${flavors_to_build}"
+
+for flavor in $flavors_to_build; do
+  ceph_codename="$(extract_ceph_codename "${flavor}")" || exit $?
```

Comment:
Ummm ... I'm not sure to be honest. The problem `|| exit $?` is solving is that when a command is run in a subshell like that, if it exits with failure, the calling script doesn't fail. So the "OR exit" will exit with the returned status code if the command itself failed. I know this works here because setting a variable does not alter the return code. I didn't see any suggestions to use an `ERR` trap when searching for solutions to this online. I'll have to look into it.